XIPs/zip-15-attachment-content-type.md

```diff
@@ -0,0 +1,60 @@
+---
+xip: 15
+title: Attachment Content Type
+description: Content type for basic attachments
+author: Pat Nakajima
+status: Review
+type: Standards Track
+category: XRC
+created: 2023-02-15
+---
+
+## Abstract
+
+This XRC proposes a new content type for messages that do not consist of text. It attempts to include only the bare minimum for clients to determine how to display such messages in order to provide flexibility for more rich types in the future.
+
+## Motivation
+
+Not everything in a conversation is text. MMS, email attachments, gifs, etc are all examples of times when communications have gone beyond words. In order to provide users a real messaging platform, XMTP needs to provide this capability.
```

Comment:
I can get my head around a generic "attachment" type, if we are to treat these like files.

I'm less confident we should be relying on MIME content type alone for content meant to be rendered inside client apps. It feels like without Content-Disposition and Content-Transfer-Encoding we are going to run into the same problems that necessitated those being added to MIME.

For media types being rendered inside an application there are likely more fields than just `mimeType` and `filename` that we would want. For example, an image type might want `size`, `alt-text`, `attribution`, etc. Cases like those might be better served with bespoke XMTP `ContentType`s

Reply:
> It feels like without Content-Disposition and Content-Transfer-Encoding we are going to run into the same problems that necessitated those being added to MIME.

I think encoding could definitely be worth adding. I'm not as clear where disposition would be as necessary here but could just be misunderstanding.

> For example, an image type might want size, alt-text, attribution, etc. Cases like those might be better served with bespoke XMTP ContentTypes

Totally agree that we could have some much richer types. I think this type would be a decent bare bones solution that gets us unblocked on a lot of functionality for now, if we ever got to a place where we had a more complete set of bespoke types, we could look at deprecating this. I think that's a much longer process though.

Reply:
Can fields like size, alt-text, etc be added as optional parameters to support richer types from the get go?

Reply:
In a future with bespoke content types for Image, Audio, Video, etc when should apps use an attachment content type vs a dedicated content type (e.g. image)?

Trying to understand if attachment type has a use case in that future. Or if it's certain that it'll be deprecated.

Reply:
I agree with @nakajima that we don't want to throw in every possible configuration to this generic content type. If we want an image type with alt-text, we should build an image content type that has the appropriate fields. That way a client can support the image content type fully or not at all.